### PR TITLE
A2 Hosting with TurboCache

### PR DESCRIPTION
**Pull request by BenjaminCool**
branding changes and functionality updates.  Now works to compliment TurboCache to prevent cart pages and user specific pages from being cached.

### CODE REVIEW

File: a2hosting.php [hunk 1]
```diff
@@ -51,7 +76,23 @@ public function __construct()
 
 	public function install()
 	{
-		return parent::install() && $this->registerHook('backOfficeHeader');
+		return parent::install() && 
+			$this->registerHook('backOfficeHeader') &&
+			$this->registerHook('header') &&
+			$this->registerHook('actionAuthentication') &&
+			$this->registerHook('actionCartSave') &&
+			$this->registerHook('displayHeader') &&
```
Comment (Quetzacoalt91) [inline a2hosting.php]: You can register header, or displayHeader, but this is useless to register both.

File: a2hosting.php [hunk 2]
```diff
@@ -51,7 +76,23 @@ public function __construct()
...
+			(bool)true;
```
Comment (Quetzacoalt91) [inline a2hosting.php:84]: This line can be removed

**Comment by Quetzacoalt91** [comment]
Hi @BenjaminCool,

Can you give me more details about these cookies used on the front office ? I guess they are related with TurboCache but I don't know how.
Meanwhile, can you please use the `Cookie` class available in the PrestaShop Core ? Thanks.

**Comment by BenjaminCool** [comment]
Hello Thomas,

Yes, I will tell you more about these cookies.

As you can see they get set when there is either an item in the users
cart or when the user is logged in.  These cookies must be read by the
web server to determine whether the page should be cached by the
server or not.  If these cookies are encoded, the web server cannot
determine if the page should be cached or not.  If you are referring
to using the cookie class for checking if the user is logged in with
isLogged() and isLoggedBack(), then yes, I can use the cookie class.

**Comment by Quetzacoalt91** [comment]
Hi @BenjaminCool,

Thanks for the explanation. Before I merge the PR, can you please update the version number and remove the hook header which is duplicated ?

**Comment by BenjaminCool** [comment]
I have removed the duplicate displayHeader and updated the version number

**Comment by Quetzacoalt91** [comment]
Hi @BenjaminCool,

Thanks, your PR has just been merged and the changes should be available from tomorrow.
For your informarion, you also had to update the config.xml file, in order to use the PrestaShop cache (done in e4f8955)

Best regards

**Pull request merged**